homeassistant/components/tibber/diagnostics.py

```diff
@@ -13,19 +13,23 @@ async def async_get_config_entry_diagnostics(
     hass: HomeAssistant, config_entry: ConfigEntry
 ) -> dict:
     """Return diagnostics for a config entry."""
-    tibber_connection: tibber.Tibber = hass.data[DOMAIN]
-
     diagnostics_data = {}
 
+    tibber_connection: tibber.Tibber = hass.data.get(DOMAIN)
```

Comment:
How can the domain be missing?

Reply:
The point of the change is not to register any data before the ConfigEntry could be initialized.
If the connection to the external API fails a ConfigEntryNotReady exception is raised.

@MartinHjelmare - I did not misunderstand from your comment https://github.com/home-assistant/core/pull/81785#pullrequestreview-1172308995, did I?

Reply:
If the config entry is not ready the tibber connection can't give any diagnostics. We currently don't seem to have a good way of handling this in our diagnostics endpoint handler. I'd like to be able to raise an exception to say that diagnostics isn't available.

Side note: We could still give general information about the config entry, config entry data etc, but that's not what the current diagnostics for tibber includes.

Reply:
My understanding is that the diagnostics information is from the Tibber integration. The integration is dependent on the API connection. If the API connection is failing users experience integration not working and may raise issues.
It can happen that there is no "homes" info but API connection was OK. 

In the future the pyTibber library should provide errors upstream so the integration can provide more diag data. But that is unfortunately not the case now.

All in all I believe it is valid and adds value to capture the API state in the diag data. It will be very obvious to see in issues where a user cannot even establish connection with the API.

Reply:
If the config entry is not ready, and we want to give connection diagnostics we should try to create a new connection here and discard if afterwards, I think. We can then say what part of the connection works and what doesn't work.

Reply:
I'm not sure if I'm getting more confused or understand things better.
Do you mean that the async_get_config_entry_diagnostics should only return something if there is diag data _on_ the config entry? So if the entry is not ready no diag data should be returned?

Reply:
No, I think we could return diagnostics data if the config entry is not ready. But we can't rely on the connection of the config entry as it's not ready. We can try to create a new connection here though and return diagnostics about it. Eg if we can't connect at all, we say that, if we can connect but not get all data we say that.

Reply:
Ok, I renamed the entry to "tibber_connection", it it's false we know the entry is not ready.
It doesn't make sense to retry a connection in the diagnostics module, feels way too much overhead.